src/Microsoft.Extensions.PlatformAbstractions/Native/NativeMethods.Darwin.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Extensions.PlatformAbstractions.Native
+{
+    internal static partial class NativeMethods
+    {
+        public static class Darwin
+        {
+            private const int CTL_KERN = 1;
+            private const int KERN_OSRELEASE = 2;
+
+            public unsafe static string GetKernelRelease()
+            {
+                // This is way to small. Using it to test errno/GetLastError
+                const uint BUFFER_LENGTH = 32;
+
+                var name = stackalloc int[2];
+                name[0] = CTL_KERN;
+                name[1] = KERN_OSRELEASE;
+
+                var buf = stackalloc byte[(int)BUFFER_LENGTH];
+                try
+                {
+                    var len = stackalloc uint[1];
+                    *len = BUFFER_LENGTH;
+                    if (sysctl(name, 2, buf, len, IntPtr.Zero, 0) != 0)
+                    {
+                        throw new PlatformNotSupportedException("Failed to get Darwin Kernel Version");
+                    }
+                    else if(*len >= BUFFER_LENGTH)
```

Comment:
if the return value is 0 and `*len` is 0 then it still spurious since you are comparing against `BUFFER_LENGTH` and not 0, no?